python/lsst/daf/butler/core/config.py

```diff
@@ -403,8 +454,16 @@ def __init__(self, other=None, validate=True, mergeDefaults=True, searchPaths=No
         externalConfig = Config(other)
 
         # Select the part we need from it
-        if self.component is not None and self.component in externalConfig:
-            externalConfig.data = externalConfig.data[self.component]
+        # To simplify the use of !include we also check for the existence of
+        # component.component (since the included files can themselves
+        # include the component name)
+        if self.component is not None:
+            doubled = "{0}.{0}".format(self.component)
+            # Must check for double depth first
+            if doubled in externalConfig:
+                externalConfig = externalConfig[doubled]
+            elif self.component in externalConfig:
+                externalConfig.data = externalConfig.data[self.component]
```

Comment:
I think in your example the right thing would happen because I strip `datastore.datastore` but the child will be `datastore.child1.datastore` and the `DatastoreConfig` constructor would be given `datastore.child1` as an argument and would correctly handle it. I was assuming chained datastores would recursively call datastore constructors with `DatastoreConfig` rather than having the parent datastore try to parse everything itself. I might be misunderstanding your point though. I was worrying this morning that `datastore.datastore` might be deliberate 

The code should already work without including the `datastore:` level in datastore config files. I'm okay with requiring that we don't have that in files that are `!include`ed but I was thinking it's clearer for people if they can see from the contents of the file that it's a datastore config rather than relying on the file name or parent directory name.

Reply:
I was just thinking that the `child1 in `datastore.child1.datastore` would be an unnecessary single-item dict that it'd be nice to strip out, especially because we *do* strip out a conceptually similar single-item dict in `datastore.datastore` -> `datastore` (i.e. I don't think a composite Datastore's config should have a different structure based on whether it calls its child "datastore" or something else).  But I suppose it's a judgement call whether that's more or less important than having datastore configs start with their headings; that's nice too.

Reply:
How about we punt this discussion until after I see how chained datastore configs really work?

Reply:
Works for me.